src/main/scala/nodes/images/FisherVector.scala

```diff
@@ -49,3 +51,45 @@ case class FisherVector(gmm: GaussianMixtureModel)
     convert(DenseMatrix.horzcat(fv1, fv2), Float)
   }
 }
+
+/**
+ * Trains a scala Fisher Vector implementation, via
+ * estimating a GMM by treating each column of the inputs as a separate
+ * DenseVector input to [[GaussianMixtureModelEstimator]]
+ *
+ * TODO: Pending philosophical discussions on how to best make it so you can
+ * swap in GMM, KMeans++, etc. for Fisher Vectors. For now just hard-codes GMM here
+ *
+ * @param k Number of centers to estimate.
+ */
+case class GMMFisherVectorEstimator(k: Int) extends Estimator[DenseMatrix[Float], DenseMatrix[Float]] {
+  protected def fit(data: RDD[DenseMatrix[Float]]): FisherVector = {
+    val gmmTrainingData = data.flatMap(x => convert(MatrixUtils.matrixToColArray(x), Double))
+    val gmmEst = new GaussianMixtureModelEstimator(k)
+    val gmm = gmmEst.fit(gmmTrainingData)
+    FisherVector(gmm)
+  }
+}
+
+/**
+ * Trains either a scala or an `enceval` Fisher Vector implementation, via
+ * estimating a GMM by treating each column of the inputs as a separate
+ * DenseVector input to [[GaussianMixtureModelEstimator]]
+ *
+ * Automatically decides which implementation to use when node-level optimization is enabled.
+ *
+ * @param k Number of centers to estimate.
+ */
+
+case class OptimizableGMMFisherVectorEstimator(k: Int) extends OptimizableEstimator[DenseMatrix[Float], DenseMatrix[Float]] {
```

Comment:
Can we not call these things `Optimizable` when they are? I really don't want users to have to think about what's going on under the covers - the fact that the system is optimizing this under the covers should be incidental to their experience.